plugins/deprecation.nix

```
@@ -64,22 +72,34 @@ in
           Please use `plugins.web-devicons` or `plugins.mini.modules.icons` with `plugins.mini.mockDevIcons` instead.
         ''
     ) iconsPackagePlugins
-    # Show a warning when web-devicons is auto-enabled
     ++ [
       (
         { config, options, ... }:
         {
-          config = lib.mkIf (options.plugins.web-devicons.enable.highestPrio == 1490) {
-            warnings = [
-              ''
-                Nixvim: `plugins.web-devicons` was enabled automatically because the following plugins are enabled.
-                This behaviour is deprecated. Please explicitly define `plugins.web-devicons.enable` or alternatively
-                enable `plugins.mini.enable` with `plugins.mini.modules.icons` and `plugins.mini.mockDevIcons`.
-                ${lib.concatMapStringsSep "\n" (name: "plugins.${name}") (
-                  builtins.filter (name: config.plugins.${name}.enable) iconsPackagePlugins
-                )}
-              ''
-            ];
+          config = {
+            warnings =
+              # Warning for auto-enabled web-devicons
+              lib.optionals (options.plugins.web-devicons.enable.highestPrio == 1490) [
+                ''
+                  Nixvim: `plugins.web-devicons` was enabled automatically because the following plugins are enabled.
+                  This behaviour is deprecated. Please explicitly define `plugins.web-devicons.enable` or alternatively
+                  enable `plugins.mini.enable` with `plugins.mini.modules.icons` and `plugins.mini.mockDevIcons`.
+                  ${lib.concatMapStringsSep "\n" (name: "plugins.${name}") (
+                    builtins.filter (name: config.plugins.${name}.enable) iconsPackagePlugins
+                  )}
+                ''
+              ]
+              ++
+                # Warnings for deprecated plugins
+                (lib.concatMap (
+                  plugin:
+                  lib.optionals (config.plugins."${plugin}".enable or false) [
+                    ''
+                      Nixvim Warning: The `${plugin}` plugin has been deprecated.
+                      ${deprecated.${plugin}}
+                    ''
+                  ]
+                ) (lib.attrNames deprecated));
```

```suggestion
                lib.concatMap (
                  plugin:
                  lib.optional (config.plugins."${plugin}".enable or false) ''
                    Nixvim Warning: The `${plugin}` plugin has been deprecated.
                    ${deprecated.${plugin}}
                  ''
                ) (lib.attrNames deprecated);
```

Two subjective suggestions:
1. Don't wrap concatenation section in (unnecessary parentheses)
2. Use `optional` (no `s`) for the optional singleton string
3. (Not in above suggestion) use [`foldAttrs`](https://noogle.dev/f/lib/foldAttrs) instead of `attrNames` + `concatMap` combo

---

2 is always hard for me here because I spent so much time in nixpkgs where it's frowned upon, lol.

---

> 2 is always hard for me here because I spent so much time in nixpkgs where it's frowned upon, lol.

As I say, subjective. Take what you agree with and discard what you don't. :slightly_smiling_face: 



> ~Two~ ~_three_~ _five (sorry)_ subjective suggestions:

I'm sorry, but I have the same bad habit or editing after posting :sob:

---

Comments felt really redundant for these 2 blocks. Just removing them.